src/reaction_network.jl

```diff
@@ -233,11 +233,12 @@ function recursive_expand_functions!(expr::ExprValues)
     (typeof(expr)!=Expr) && (return expr)
     foreach(i -> expr.args[i] = recursive_expand_functions!(expr.args[i]), 1:length(expr.args))
     if expr.head == :call
-        haskey(funcdict, expr.args[1]) && return funcdict[expr.args[1]](expr.args[2:end])
         in(expr.args[1],hill_name) && return hill(expr)
         in(expr.args[1],hillR_name) && return hillR(expr)
         in(expr.args[1],mm_name) && return mm(expr)
         in(expr.args[1],mmR_name) && return mmR(expr)
+
+        !isdefined(Catalyst,expr.args[1]) && (expr.args[1] = esc(expr.args[1]))
```

Comment:
the answer was this esc?

Reply:
Seems to work for the simple examples I’ve tested. Not merging yet as I’d like to try a couple more complicated expressions to double check tomorrow, including your NN example.